### ISSUE TITLE
When `disable: true` on `generate` block after it has been enabled, cached file is not removed

### ISSUE THREAD

**Issue by fe-ax** [gruntwork-io/terragrunt#3096]
**Describe the bug**
When a block is generated, it's not removed when the `generate` block is disabled. Manual cache invalidation is needed.

**To Reproduce**
First run this Terragrunt config

```hcl
generate "test" {
  disable = false
  path = "test.tf"
  if_exists = "overwrite"
  contents = <
output "test" {
  value = "test"
}
EOT
}
 
terraform {
  source = "./tf/"
}
```

terragrunt plan

```
marco@test$ terragrunt plan
Initializing the backend...
 
Initializing provider plugins...
 
Terraform has been successfully initialized!
 
...
 
Changes to Outputs:
  + test = "test"
 
...
```

Set to disable

```hcl
generate "test" {
  disable   = true
  path      = "test.tf"
  if_exists = "overwrite"
  contents  = <<EOT
output "test" {
  value = "test"
}
EOT
}

terraform {
  source = "./tf/"
}
```

Nothing changed:

```
marco@test$ terragrunt plan
Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

...

Changes to Outputs:
  + test = "test"

...
```

Flushing `.terragrunt-cache` cache, and with it the generated [test.tf](http://test.tf)

```
marco@test$ rm -rf ./.terragrunt-cache
marco@test$ terragrunt plan
INFO[0000] Downloading Terraform configurations from file:///home/marco/notes/testtf/tf into /home/marco/notes/testtf/.terragrunt-cache/wfIPpErBuUByitLEL5eelRC_g0s/vBTqYIOukLa735JrmTnCK9zPHsY 

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

No changes. Your infrastructure matches the configuration.

...
```

**Expected behavior**
Removal of the generated file if it has the signature.

**Nice to have**

- [X] Terminal output
- [ ] Screenshots

**Versions**

* Terragrunt version: v0.55.13
* Terraform version:
* Environment details (Ubuntu 20.04, Windows 10, etc.): WSL ubuntu 22.04

**Additional context**
[The disable feature PR](https://github.com/gruntwork-io/terragrunt/pull/2497)

**Workaround**
For now you can use the following to clear the contents of the file and leave an empty file when disabled:

```hcl
generate "test" {
  #disable = true
  path = "test.tf"
  if_exists = "overwrite"
contents = !local.enabled_test_tf ? "" : <
  value = "test"
}
EOT
}
 
terraform {
  source = "./tf/"
}
```

**Comment by levkohimins** [comment]
Resolved in [v0.57.5](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.57.5) release.

Added `if_disabled` attribute for `generate` block. Make sure to read [docs].(https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#generate)

Using your example:

```hcl
generate "test" {
  disable = false
  if_disabled = "remove"
  path = "test.tf"
  if_exists = "overwrite"
  contents = <<EOF
output "test" {
  value = "test"
}
EOF
}
```

**Comment by fe-ax** [comment]
Works like a charm. Thanks!
